server/tinylicious/src/test/dbFactory.spec.ts

```diff
@@ -0,0 +1,167 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+import * as fs from "fs";
+import * as os from "os";
+import * as path from "path";
+import { ICollection, IDb } from "@fluidframework/server-services-core";
+import { Provider } from "nconf";
+import { DbFactory } from "../services";
+
+/**
+ * JSON value stored within the database collections used in the test
+ */
+interface IValue {
+    _id?: number;
+    foo?: string;
+    a?: {
+        b: number;
+    }
+}
+
+/**
+ * DB factory test configuration
+ */
+interface ITestConfig {
+    // config.json values used as part of the test
+    value: {
+        db: {
+            inMemory: boolean;
+            path?: string;
+        }
+    }
+
+    // dispose method to allow the config to do any cleanup - i.e. delete the temporary
+    // directory used by leveldb
+    dispose: () => void;
+}
+
+/**
+ * Factory to create per database configurations for the tests
+ */
+interface IConfigFactory {
+    name: string;
+    create: () => ITestConfig;
+}
+
+describe("Tinylicious", () => {
+    describe("Services", () => {
+        const configFactories = new Array<IConfigFactory>(
+            {
+                name: "inMemoryCollection",
+                create: () => ({
+                    value: {
+                        db: {
+                            inMemory: true,
+                        },
+                    },
+                    dispose: () => { },
+                }),
+            },
+            {
+                name: "levelDb",
+                create: () => {
+                    const levelDir = fs.mkdtempSync(path.join(os.tmpdir(), "level-"));
+
+                    return {
+                        value: {
+                            db: {
+                                inMemory: true,
```

Comment:
false?

Reply:
Ah - good catch! Fixed things up to actually turn leveldb on...

As part of this also needed to tweak the tests a bit. MongoDB is the basis for these DB APIs. And both in memory/level are subsets of the Mongo API. Tinylicious/r11s obviously don't use the full set of functionality.

The test now better captures the usage of each collection and serves as a bit of a conformance test should anyone add another use case.

Having a more constrained DB API may be nice to better highlight the actual required functionality.